packages/vscode-extension/src/commonlib/m365Login.ts

```diff
@@ -270,6 +270,21 @@ export class M365Login extends BasicLogin implements M365TokenProvider {
       );
       if (tokenRes.isOk()) {
         const tokenJson = ConvertTokenToJson(tokenRes.value);
+        // if token is empty, try to get token by app studio scope, normally this should only affect UX
```

Comment:
What if always get app studio token for `getStatus` (remove the `tokenRequest` input)?

It's a potential issue if input scope if A but the returned token is for scope B.

Reply:
Or, just return empty `accountInfo` but the token matches input scope.

Reply:
Like `getTeamsAppInternalId` will try to get graph scope token, I think always returning appStudio token is not a safe way.

Reply:
So returning appStudio token for empty json is also unsafe. I prefer just returning empty tokenjson, but the token itself is still the original scope.

Reply:
For now, it's not a good idea to return the original token. UX is using this to get sideloading permission and user information.
https://github.com/OfficeDev/TeamsFx/blob/bcfc8de2efa4685b79a59fb424acc1836d7c8aa7/packages/vscode-extension/src/treeview/account/accountTreeViewProvider.ts#L78
![image](https://github.com/OfficeDev/TeamsFx/assets/13211513/53786184-76a0-4352-9650-61008dc7a3e9)

Reply:
Who calls the `m365AccountStatusChangeHandler`? Is it possible to always pass appstudio token to `m365AccountStatusChangeHandler`?

UX fix can be a wrapper with hard-coding logic. But for the general `getStatus` function, keep consistent behavior.

Reply:
[Here](https://github.com/OfficeDev/TeamsFx/blob/dev/packages/vscode-extension/src/treeview/account/accountTreeViewProvider.ts#L36) passes appstudio scope, then why `m365AccountStatusChangeHandler` receives token for other scope?

Reply:
I mean if I change this to return origin scope token and appStudio account info.

Reply:
I see. So for `getStatus`, keep the behavior meaningful - return token and json always matching the input scope. Since `getStatus` is a public api, we cannot assume all callers need the appstudio fallback.

For UX, better to fix in `setStatusChangeMap` - currently it seems ignore the input `tokenRequest` when notifying status.

Reply:
I will have two more prs. One will focus on saving the state for notification, while the other will involve using basic scope for login and updating getToken to utilize request scope.

Reply:
And may roll the `getStatus` workaround back in the notify fix~